### DISCUSSION TITLE
Overzooming with upsampling

### DISCUSSION THREAD

**Discussion by Plantain** [developmentseed/titiler#460] [Q&A]
#### Problem description
Zooming past the maximum level of a COG should upsample using the specified resampling= algorithm, instead it seems to do a simple scale without any upsampling occurring.

Requesting tiles with:
https://localhost:8080/tiles/9/473/293@2x?resampling=lanczos

Zoom level 8 (native resolution for this COG)
![image](https://user-images.githubusercontent.com/319895/164166537-5e55816e-8caa-4c9d-b962-adc9be9d214a.png)

Zoom level 9 (max zoom +1)
![image](https://user-images.githubusercontent.com/319895/164167155-ae9b3788-dbf4-4832-a88a-64b9dc3413fd.png)

Notice the overzoomed image has large blocks - no upsampling is happening.

I'm not sure if this is a bug/feature request for titiler, or a limitation of rasterio?

Original file is at http://static2.skysight.io/demo.tiff

**Comment by Plantain** [comment]
![ex](https://user-images.githubusercontent.com/319895/164167977-9b92369a-a2c5-409c-9fb5-71f5872dd7c7.png)
zoomed in further to make it more obvious...

**Comment by vincentsarago** [comment]
@Plantain I'm sorry but I'm not seeing anything wrong here. I'll translate this issue as discussion because this is not really a bug (or if there is one it would be either on rio-tiler or rasterio)

Here is what I get for tile `9/473/293` with 1024x1024 tile size
![image](https://user-images.githubusercontent.com/10407788/165079031-6da8b611-eb73-4ead-aeeb-5fcb3291c49d.jpeg)

**Reply by Plantain** [reply]
Ok - but I you set resampling=bilinear, I would not expect to see these 'pixel' blocks if it was upsampled with bilinear.

**Reply by vincentsarago** [reply]
here is what I get with bilinear 
![image](https://user-images.githubusercontent.com/10407788/165082812-ecbb0cab-3fa7-4e1b-a29a-b4fe5a16dfc8.jpeg)

It's not clear to me what the `correct` result for you would be. Ideally it would help if you could produce the expected result so we can compare.

**Reply by Plantain** [reply]
Interesting, I'm not sure why I am not getting that when I add &resampling=bilinear, I get the same blocky image - possibly a terrible bug in my code, let me dig further.

I do still see 'pixel' blocks in your example though, my expectation was that with bilinear or basically any interpolation other than nearest, that there would never be blocky rendering

**Reply by lenalebt** [reply]
Hey, were you able to solve this @Plantain ? I am facing the same thing :smile: .

**Reply by lenalebt** [reply]
Ah, you know how this goes: The minute you ask someone, you find a solution. I was reprojecting as well, and did not really notice. When I started using `reproject=bilinear` instead of `resampling=bilinear`, it worked.
